kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/ServiceControlCli.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.ctl
+
+import org.apache.curator.utils.ZKPaths
+
+import org.apache.kyuubi.Logging
+import org.apache.kyuubi.ha.HighAvailabilityConf._
+import org.apache.kyuubi.ha.client.ServiceDiscovery
+
+private[ctl] object  KyuubiCtlAction extends Enumeration {
+  type KyuubiCtlAction = Value
+  val CREATE, GET, DELETE, LIST, HELP = Value
+}
+
+private[ctl] object KyuubiCtlActionService extends Enumeration {
+  type KyuubiCtlActionService = Value
+  val SERVER, ENGINE = Value
+}
+
+/**
+ * Main gateway of lauching a Kyuubi Ctl action.
+ * See usage in [[KyuubiCtlArguments.printUsageAndExit]].
+ */
+private[kyuubi] class ServiceControlCli extends Logging {
+  import ServiceDiscovery._
+
+  def doAction(args: Array[String]): Unit = {
+    val ctlArgs = parseArguments(args)
+    if (ctlArgs.verbose) {
+      info(ctlArgs.toString)
+    }
+    ctlArgs.action match {
+      case KyuubiCtlAction.CREATE => create(ctlArgs)
+      case KyuubiCtlAction.HELP => printUsage(ctlArgs)
+      // TODO: support other actions
+      case action => throw new UnsupportedOperationException(s"Unsupported action type:$action")
+    }
+  }
+
+  protected def parseArguments(args: Array[String]): KyuubiCtlArguments = {
+    new KyuubiCtlArguments(args)
+  }
+
+  /**
+   * Expose Kyuubi server instance to another domain.
+   */
+  private def create(args: KyuubiCtlArguments): Unit = {
+    val defaultConf = args.defaultKyuubiConf.clone
```

Comment:
we don't need clone here

the Kyuubi arguments take precedence over the loaded ones from the file.

we can simply set the argument to the `kyuubiConf`

Reply:
ok

Reply:
the conf loading logic can be simply implemented like this:

1. val kyuubiConf = KyuubiConf() to load configs from the config file
2. parse arguments
3. set the parsed kv to `kyuubiConf` if there is a defined key (no need to declare a field for them, e.g. namespace)
4. keep a field in KyuubiCtlArguments if no pre-defined key in kyuubiconf, e.g. namespace
5. verify the kv if necessary, (I don't think there are all need explicitly verifying logic, which could be delay to zk client)
6. create zk client by conf
7. operate znode via zk client and other parameters in KyuubiCtlArguments

Reply:
The parsed kv could be set into kyuubiConf are `zk quorum` and `zk namespce`.

But for create operation, 
the zk quorum/namespace specified might is different with that of default kyuubi conf.
So, I can not set the parsed zk quorum to `kyuubiConf`.

Reply:
For other operations(get, delete, list) so far, we can set the parsed zk args into kyuubiconf.

Reply:
We can set the parsed kv to kyuubiConf during args validation phase.